docs/CONTRIBUTING.md

```diff
@@ -35,6 +35,58 @@ The source code of ABACUS is based on several modules. Under the ABACUS root dir
 - `tests`: End-to-end test cases;
 - `tools`: the script for generating the numerical atomic orbitals.
 
+For those who are interested in the source code, the following figure shows the structure of the source code.
+
+```
+|-- module_base                 A basic module including 
+|   |                           (1) Mathematical library interface functions: BLAS, LAPACK, Scalapack;
+|   |                           (2) Custom data classes: matrix, vector definitions and related functions;
+|   |                           (3) Parallelization functions: MPI, OpenMP;
+|   |                           (4) Utility functions: timer, random number generator, etc.
+|   |                           (5) Global parameters: input parameters, element names, mathematical and physical constants.
+|   |-- module_container        The container module for storing data and performing operations on them and on different architectures.
+|-- module_basis                Basis means the basis set to expand the wave function.
+|   |-- module_ao               Atomic orbital basis set to be refactored.
+|   |-- module_nao              New numerical atomic orbital basis set for two-center integrals in LCAO calculations
+|   `-- module_pw               Plane wave basis set for PW calculations
```

Comment:
for presently I spend most of time on this module, a more appropriate description might be "For plane wave involved calculations", because in esolver_fp class, we already have PW_Basis* pw_rho to calculate electronic density, and it is, really called in both esolver_ks_pw and esolver_ks_lcao.